panels/region/cc-input-chooser.c

```diff
@@ -1211,3 +1211,30 @@ cc_input_chooser_get_selected (GtkWidget  *chooser,
 
   return TRUE;
 }
+
+void
+cc_input_chooser_set_is_small_screen (GtkWidget *chooser,
+                                      gboolean   is_small_screen)
+{
+        CcInputChooserPrivate *priv = GET_PRIVATE (chooser);
+
+        if (is_small_screen) {
+                guint width, height;
+
+                gtk_window_get_size (gtk_window_get_transient_for (GTK_WINDOW (chooser)), &width, &height);
+                gtk_widget_set_size_request (GTK_WIDGET (chooser), width * MAIN_WINDOW_WIDTH_RATIO, height);
```

Comment:
I personally prefer to stay with the transient's geometry. Using the scrolled-window's min_content_width & min_content_height would mean that we'll have to use arbitrary hardcoded sizes. Also, setting the scrolledwindow's size won't take into account the buttons and other misc margins and paddings.

By setting the dialog's size-request, we avoid all the previous issues and guarantee that everything is visible from a small screen.